mbtools/models.py

```diff
@@ -459,6 +459,22 @@ def get_elements_with_string_in_class(self, class_string):
         xpath_query = f"//*[contains(@class, '{class_string}')]"
         return self.etree_fragments[0].xpath(xpath_query)
 
+    def get_elements_with_exact_class(self, classes):
+        # NOTE This method returns a serites of elements whose classes match
+        #   exactly one of the classes provided
+        elems = []
+        for item in classes:
+            xpath_query = f"//div[@class='{item}']"
+            elems.extend(self.etree_fragments[0].xpath(xpath_query))
+        return elems
+
+    def add_ib_content_id(self, classes_to_inject):
+        elems = self.get_elements_with_exact_class(classes_to_inject)
+        for elem in elems:
+            if 'data-content-id' not in elem.attrib.keys():
+                elem.attrib['data-content-id'] = str(uuid4())
+        self.update_html()
```

Comment:
This code will get moved into your script vs in the models, but when doing that you can also probably consolidate this to avoid two functions by using bs4's class search API per [here](https://beautiful-soup-4.readthedocs.io/en/latest/#searching-by-css-class). That would also help you avoid the same queue vs iterate issue mentioned elsewhere in this PR.